lib/runnable.js

```diff
@@ -277,4 +275,18 @@ Runnable.prototype.run = function(fn){
       done();
     }
   }
+
+  function callFnAsync(fn) {
+    fn.call(ctx, function(err){
+      if (err instanceof Error || toString.call(err) === "[object Error]") return done(err);
```

Comment:
I didn't write this.. the lines were just moved into the method named `callFnAsync`.

I can make the changes if you like.